### ISSUE TITLE
the best models  case insensitive?

### ISSUE THREAD

**Issue by xiaoyubing** [clovaai/deep-text-recognition-benchmark#7]
the best models  case insensitive?can you give case sensitive best models?

**Comment by ku21fan** [comment]
Yes. the model is case-insensitive. 
We will give case-sensitive models soon (maybe in 1 week?).

**Comment by ku21fan** [comment]
I just uploaded the case-sensitive model. please check [here](https://drive.google.com/drive/folders/15WPsuPJDCzhp2SvYZLRj8mAlT3zmoAMW)
But it could have low accuracy because the training data and validation data contains some case insensitive labeling. 
In detail:

1. Some fonts always generate the upper case even though the ground truth label have lower cases. Some example of MJSynth or SynthText might have been generated by these fonts.
2. The training data of IIIT5k and SVT are used as the validation data.  Their label is always the upper case, even though the ground truth label has the lower case. please check [here](https://drive.google.com/file/d/18EqkfVFTWNVFczJ6tuKQMx-FZTmxcurR/view?usp=sharing)

If you want a more accurate case sensitive model, fine-tuning this model with your own data or more cleaned case sensitive data.

Best.

**Comment by xiaoyubing** [comment]
thanks

**Comment by ku21fan** [comment]
I forgot to notify one thing.
use case-sensitive model with `--sensitive` option.

**Comment by solitaire2015** [comment]
@ku21fan Thanks for your great job. I have a question:
I test the pre-trained model of case-sensitive on IIIT5K 3000 by filtering special characters and change to upper case, I got a good accuracy: 90%. I wonder if the model is trained only on MJSynth + SYNText? The performance is incredible. Could you please explain why? 

Best.

**Comment by ku21fan** [comment]
@solitaire2015 
Yes, the model is trained only on MJSynth + SynthText. 
But, in case `--sensitive` mode, we used **_all_** of SynthText.

After [validation checks](https://github.com/clovaai/deep-text-recognition-benchmark/issues/6#issuecomment-487453283) for each bounding box,
[a] SynthText has about 5.5 million images which do not contain special characters (only alphanumeric).
[b] SynthText has about 1.5 million images which contain special characters. 

Our model was trained on [a] with case insensitive mode,
and trained on [a] and [b] with `--sensitive` mode
For now, only [a] is uploaded in our google / baidu drive, and I will upload [b] soon.

Best.

**Comment by solitaire2015** [comment]
@ku21fan That's great, thank you very much.

**Comment by ku21fan** [comment]
@solitaire2015 I just uploaded [b] as 'ST_spe.zip' at [here](https://drive.google.com/drive/folders/192UfE9agQUMNq6AgU3_E05_FcPZK4hyt)
Download/unzip/move it into the `data_lmdb_release/training/`
And then try to train with ST_spe :)

**Comment by solitaire2015** [comment]
@ku21fan I trained the model on `--sensitive` mode with the [a] + [b]. here are the training accuracy and loss:
    [272000/300000] Loss: 0.06758 elapsed_time: 135756.64656
acer                , gt: acer                ,   True
**Endless             , gt: ENDLESS             ,   False**
BBQ                 , gt: BBQ                 ,   True
The                 , gt: The                 ,   True
THE                 , gt: THE                 ,   True
[272000/300000] valid loss: 2.01074 accuracy: 64.745, norm_ED: 1728.75
best_accuracy: 66.390, best_norm_ED: 1609.72

I trained this model on two 1080Ti, selected the best combination, TPS + ResNet + BiLSTM + Attn, for some reasons I add one convolutional layer at the end of ResNet feature extractor to keep the output width to be 25(26 before), I test the model on IIIT5k 3000, after filtering special characters and converting lower case to upper case, I got total accuracy: 88.5%, a little bit lower than yours 90%.

I investigate some failure cases of IIIT5K, some special characters are treated as alphanumeric characters, I think it's the reason why the sensitive model is more accurate than the insensitive model.

Best.

**Comment by alwc** [comment]
Hi @solitaire2015, did you use grayscale or rgb mode for your input images?

**Comment by solitaire2015** [comment]
@alwc I use grayscale images.

**Comment by mahendra047** [comment]
Hi,
I trained the model on --sensitive mode with the  MJSynth + ST_spe . here are the training accuracy and loss:
[79000/300000] Loss: 0.09761 elapsed_time: 103154.51228
FOR,                , gt: FOR                 ,   False
Mo-vediplon         , gt: MOVCD1PLO1          ,   False
REBATE              , gt: REBATE              ,   True
WALK                , gt: WALK                ,   True
NISSAN-SBI          , gt: NISSANSBI           ,   False
[79000/300000] valid loss: 2.17037 accuracy: 58.939, norm_ED: 2006.77
best_accuracy: 60.812, best_norm_ED: 1922.32

**model gives constant accuracy from this iteration **

[47500/300000] valid loss: 2.29950 accuracy: 56.822, norm_ED: 2124.85
best_accuracy: 60.812, best_norm_ED: 1922.32
[48000/300000] Loss: 0.11939 elapsed_time: 62696.96234
proudly             , gt: proudly             ,   True
LOVE                , gt: LOVE                ,   True
SAXONS              , gt: SAXONS              ,   True
TESCO               , gt: TESCO               ,   True

------------ Options -------------
experiment_name: TPS-ResNet-BiLSTM-Attn-Seed1111
train_data: data_lmdb_release/training
valid_data: data_lmdb_release/validation
manualSeed: 1111
workers: 4
batch_size: 192
num_iter: 300000
valInterval: 500
continue_model:
adam: False
lr: 1
beta1: 0.9
rho: 0.95
eps: 1e-08
grad_clip: 5
select_data: ['MJ', 'ST_spe']
batch_ratio: ['0.5', '0.5']
total_data_usage_ratio: 1.0
batch_max_length: 25
imgH: 32
imgW: 100
rgb: False
character: 0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ!"#$$
sensitive: True
PAD: False
data_filtering_off: False
Transformation: TPS
FeatureExtraction: ResNet
SequenceModeling: BiLSTM
Prediction: Attn
num_fiducial: 20
input_channel: 1
output_channel: 512
hidden_size: 256
num_gpu: 1
num_class: 96
---------------------------------------
should i stop training ... 
what should i change to  improve accuracy.
